### PR TITLE
Add example recipes for CPU and XPU

### PR DESCRIPTION
**Pull request by zxd1997066**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [X] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

We verified LoRA single device finetuning for below models on both CPU and XPU. Add example recipes for them to show the support of CPU and XPU. Firstly, we added example recipes for Llama3.2 3B LoRA single device finetuning. If it is OK, we can add example recipes for remaining models.
| Model                                         | Sizes     |
|-----------------------------------------------|-----------|
| [Llama3.2-Vision](https://www.llama.com/docs/model-cards-and-prompt-formats/llama3_2#-llama-3.2-vision-models-(11b/90b)-)    | 11B |
| [Llama3.2](https://www.llama.com/docs/model-cards-and-prompt-formats/llama3_2)    | 3B |
| [Llama3.1](https://llama.meta.com/docs/model-cards-and-prompt-formats/llama3_1)    | 8B |
| [Llama3](https://llama.meta.com/llama3)    | 8B |
| [Llama2](https://llama.meta.com/llama2/)   | 7B |
| [Code-Llama2](https://ai.meta.com/blog/code-llama-large-language-model-coding/)   | 7B |
| [Mistral](https://huggingface.co/mistralai)   | 7B |
| [Gemma](https://huggingface.co/collections/google/gemma-release-65d5efbccdbb8c4202ec078b)   | 7B |
| [Microsoft Phi3](https://huggingface.co/collections/microsoft/phi-3-6626e15e9585a200d2d761e3) | Mini |
| [Qwen2](https://qwenlm.github.io/blog/qwen2/) | 1.5B |

#### Changelog
What are the changes made in this PR?
* Add example recipes for CPU and XPU, and register them in torchtune/_recipe_registry.py

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [X] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [X] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by SalmanMohammadi** [comment]
Hey @zxd1997066. Thanks so much for opening this. It looks like the main addition in these configs is updating `output_dir` and `device`. I wonder if it might be overkill to to add entire configs just to change the `device` arg, particularly if it's not as easy to maintain these configs without access to XPU-enabled hardware. What are your thoughts?

In the spirit of this PR, I think we can do more to surface our multi-device support. Perhaps we could add a section to our README or our docs to show how you can use a CLI override to easily change the device (e.g. `device=xpu`)?

**Comment by joecummings** [comment]
> Hey @zxd1997066. Thanks so much for opening this. It looks like the main addition in these configs is updating `output_dir` and `device`. I wonder if it might be overkill to to add entire configs just to change the `device` arg, particularly if it's not as easy to maintain these configs without access to XPU-enabled hardware. What are your thoughts?
> 
> In the spirit of this PR, I think we can do more to surface our multi-device support. Perhaps we could add a section to our README or our docs to show how you can use a CLI override to easily change the device (e.g. `device=xpu`)?

Seconded! We want to be careful about how many recipes we maintain in our core library. I'd love if you wanted to add a comment to the configs (that you've tested for), similar to how we outline options for our memory optimization techniques. E.g.

```yaml
device: cuda  # can also use xpu, npu, or cpu
```

And then we can add a highlight to our documentation, as well. How does this sound?

**Comment by zxd1997066** [comment]
Hi @SalmanMohammadi @joecummings , thanks for the suggestions, they sound good, and we will take a look at them.

**Comment by joecummings** [comment]
Closing as there has been no activity for awhile.
